hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/traceability/SolidityAction.java

```diff
@@ -52,6 +57,52 @@ public SolidityAction(
         this.callDepth = callDepth;
     }
 
+    /**
+     * Check that the fields are set correctly for the semantics of the underlying protobuf
+     *
+     * (Respecting, for example, protobuf `oneof` unions...)
+     */
+    public boolean isValid() {
+        boolean ok = true;
+        ok &= null != getCallType() && ContractActionType.NO_ACTION != getCallType();
+        ok &= 1 == countNonNulls(getCallingAccount(), getCallingContract());
+        ok &= null != getInput();
+        ok &= 1 == countNonNulls(getRecipientAccount(), getRecipientContract(), getInvalidSolidityAddress());
```

Comment:
Would now have to remove this case for recipients

Reply:
done